src/test/agent_test.rs

```diff
@@ -123,3 +121,120 @@ fn connection_reuse() {
     }
     assert_eq!(resp.status(), 200);
 }
+
```

Comment:
Your test setup is just great (so simple and to the point!), I think you should break out your tests into a separate test file (`src/test/timeout.rs`?)